File: youlexuan-sellergoods-interface/src/main/java/com/offcn/sellergoods/service/BrandService.java
package com.offcn.sellergoods.service;

import com.offcn.entity.PageResult;
import com.offcn.pojo.TbBrand;

import java.util.List;

public interface BrandService {

    /*查询所有*/
    public List<TbBrand> findAll();

    /*分页查询*/
    public PageResult findPage(int pageNo, int pageSize);

    /*增加品牌*/
    public void add(TbBrand brand);

    //根据id查询
    public TbBrand findOne(Long id);

    //修改
    public void update(TbBrand brand);

    //根据id集合删除
    public void delete(Long[] ids);

    //根据条件分页查询
    public PageResult findPage(TbBrand brand, int pageNo, int pageSize);
}

File: youlexuan-manager-web/src/main/webapp/js/service/brandService.js
app.service('brandService',function ($http) {

    this.findAll=function () {
        return $http.get('../brand/findAll.do');
    }
    this.findPage=function (page,rows) {
        return $http.get('../brand/findPage.do?page='+page+"&rows="+rows);
    }
    this.add=function (brand) {
        return $http.post("../brand/add.do", brand);
    }
    this.update=function (brand) {
        return $http.post("../brand/update.do", brand);
    }
    this.findOne=function (id) {
        return $http.get('../brand/findOne.do?id='+id);
    }
    this.dele=function (selectIds) {
        return $http.get('../brand/dele.do?ids='+selectIds);
    }
    this.search=function (page, rows, searchEntity) {
        return $http.post('../brand/search.do?page='+page+"&rows="+rows, searchEntity);
    }
});
















